src/dolphin/geckoCode.ts

```diff
@@ -0,0 +1,141 @@
+import { IniFile } from "./iniFile";
+
+export interface GeckoCode {
+  name: string;
+  creator: string | null;
+  notes: string[];
+  codeLines: string[];
+  enabled: boolean;
+  defaultEnabled: boolean;
+  userDefined: boolean;
+}
+
+// this is very similar to LoadCodes in GeckoCodeConfig.cpp, but skips the address and data because we don't need them
+export function loadGeckoCodes(globalIni: IniFile, localIni?: IniFile): GeckoCode[] {
+  const gcodes: GeckoCode[] = [];
+  [globalIni, localIni].forEach((ini) => {
+    if (ini === undefined) {
+      return;
+    }
+    const lines: string[] = ini.getLines("Gecko", false).filter((line) => {
+      return line.length !== 0 || line[0] !== "#";
+    });
+    let gcode: GeckoCode = {
+      name: "",
+      creator: "",
+      enabled: false,
+      defaultEnabled: false,
+      userDefined: ini === localIni,
+      notes: [],
+      codeLines: [],
+    };
+
+    lines.forEach((line) => {
+      switch (line[0]) {
+        // code name
+        case "$": {
+          if (gcode.name.length > 0) {
+            gcodes.push(gcode);
+          }
+          line = line.slice(1); // cut out the $
+
+          const creatorMatch = line.match(/\[(.*?)\]/); // searches for brackets, catches anything inside them
+          const creator = creatorMatch !== null ? creatorMatch[1] : creatorMatch;
+          const name = creator ? line.split("[")[0] : line;
+
+          gcode = {
+            ...gcode,
+            name: name,
+            creator: creator,
+            notes: [],
+            codeLines: [],
+          };
+          break;
+        }
+        // comments
+        case "*": {
+          gcode.notes.push(line.slice(1));
+          break;
+        }
+        default: {
+          gcode.codeLines.push(line);
+        }
+      }
+    });
+    if (gcode.name.length > 0) {
+      gcodes.push(gcode);
+    }
+
+    //update enabled flags
+
+    _readEnabledAndDisabled(ini, gcodes);
+
+    //set default enabled
+    if (ini === globalIni) {
+      gcodes.forEach((gcode) => {
+        gcode.defaultEnabled = gcode.enabled;
+      });
+    }
+  });
+  return gcodes;
+}
+
+function _readEnabledOrDisabled(iniFile: IniFile, section: string, enabled: boolean, codes: GeckoCode[]) {
```

Comment:
Can we refactor these to not have the `_` prefix too? They aren't private methods so it's fine to leave them without the prefix.